source/host_resolver.c

```diff
@@ -583,6 +709,112 @@ static bool s_host_entry_finished_pred(void *user_data) {
     return entry->state == DRS_SHUTTING_DOWN;
 }
 
+/* Move all of the listeners in the listener entry to the resolver thread owned list. */
+/* Assumes resolver_lock is held. */
+static void s_resolver_thread_handle_pending_listeners(
+    struct default_host_resolver *resolver,
+    const struct aws_string *host_name,
+    struct aws_linked_list *listener_list) {
+
+    AWS_ASSERT(resolver);
+    AWS_ASSERT(host_name);
+    AWS_ASSERT(listener_list);
+
+    struct host_listener_entry *listener_entry = NULL;
+    struct host_listener *listener = s_pop_host_listener_from_entry(resolver, host_name, &listener_entry);
+
+    while (listener != NULL) {
+        /* Flag this listener as in-use by the resolver thread so that it can't be destroyed from outside of that
+         * thread. */
+        listener->owned_by_resolver_thread = true;
+
+        aws_linked_list_push_back(listener_list, &listener->node);
+
+        listener = s_pop_host_listener_from_entry(resolver, host_name, &listener_entry);
+    }
+}
+
+/* Notify all listeners with resolve address callbacks, and also clean up any that are waiting to be cleaned up. */
+/* Assumes no lock is held. */
+static void s_resolver_thread_update_listeners(
```

Comment:
maybe it will be more clear to comment that the `listener_list` is a local list, so it doesn't need to be protected by a lock?

Reply:
Will do!